APMrover2/control_modes.cpp

```diff
@@ -110,53 +107,82 @@ void Rover::reset_control_switch()
     read_control_switch();
 }
 
-// read at 10 hz
-// set this to your trainer switch
-void Rover::read_trim_switch()
+// ready auxiliary switch's position
+aux_switch_pos Rover::read_aux_switch_pos()
+{
+    uint16_t radio_in = channel_aux->get_radio_in();
+    if (radio_in < AUX_SWITCH_PWM_TRIGGER_LOW) return AUX_SWITCH_LOW;
+    if (radio_in > AUX_SWITCH_PWM_TRIGGER_HIGH) return AUX_SWITCH_HIGH;
+    return AUX_SWITCH_MIDDLE;
+}
+
+// initialise position of auxiliary switch
+void Rover::init_aux_switch()
+{
+    aux_ch7 = read_aux_switch_pos();
+}
+
+// read ch7 aux switch
+void Rover::read_aux_switch()
 {
+    // do not consume input during rc or throttle failsafe
+    if (failsafe.bits & FAILSAFE_EVENT_THROTTLE || failsafe.bits & FAILSAFE_EVENT_RC) {
```

Comment:
I would add more parenthesis too for safety ((failsafe.bits & FAILSAFE_EVENT_THROTTLE) || (failsafe.bits & FAILSAFE_EVENT_RC))